source/common/router/router.h

```diff
@@ -235,7 +239,39 @@ class FilterConfig {
     for (const auto& upstream_log : config.upstream_log()) {
       upstream_logs_.push_back(AccessLog::AccessLogFactory::fromProto(upstream_log, context));
     }
+    if (config.upstream_http_filters_size() > 0) {
+      auto& server_factory_ctx = context.getServerFactoryContext();
+      const Http::FilterChainUtility::FiltersList& upstream_http_filters =
+          config.upstream_http_filters();
+      std::shared_ptr<Http::UpstreamFilterConfigProviderManager> filter_config_provider_manager =
+          Http::FilterChainUtility::createSingletonUpstreamFilterConfigProviderManager(
+              server_factory_ctx);
+      std::string prefix = context.scope().symbolTable().toString(context.scope().prefix());
+      upstream_ctx_ = std::make_unique<Upstream::UpstreamHttpFactoryContextImpl>(
+          server_factory_ctx, context.initManager(), context.scope());
+      Http::FilterChainHelper<Server::Configuration::UpstreamHttpFactoryContext,
+                              Server::Configuration::UpstreamHttpFilterConfigFactory>
+          helper(*filter_config_provider_manager, server_factory_ctx, *upstream_ctx_, prefix);
+      helper.processFilters(upstream_http_filters, "router upstream http", "router upstream http",
+                            upstream_http_filter_factories_);
+    }
+  }
+
+  bool createFilterChain(Http::FilterChainManager& manager, bool) const override {
+    if (upstream_http_filter_factories_.empty()) {
```

Comment:
why are we ignoring only_create_if_configured?

Reply:
The router default is an empty filter list -- there's nothing to do if the chain is not configured.

Reply:
nothing to do today.  how about a comment and an assert in case that changes?